openerp/addons/test_new_api/tests/test_one2many.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+from openerp.tests.common import TransactionCase
+
+
+class One2manyCase(TransactionCase):
+    def setUp(self):
+        super(One2manyCase, self).setUp()
+        self.Line = self.env["test_new_api.multi.line"]
+        self.multi = self.env["test_new_api.multi"].create({
+            "name": "What is up?"
+        })
+
+    def tearDown(self):
+        try:
+            # Check the lines first
+            self.assertItemsEqual(
+                self.multi.lines.mapped('name'),
+                map(str, range(10)))
+            # Modify the first line and drop the last one
+            self.multi.lines[0].name = "hello"
+            self.multi.lines = self.multi.lines[:-1]
+            self.assertEqual(len(self.multi.lines), 9)
+            self.assertIn("hello", self.multi.lines.mapped('name'))
+            # Invalidate the cache and check again; this crashes if the value
+            # of self.multi.lines in cache contains new records
+            self.multi.invalidate_cache()
+            self.assertEqual(len(self.multi.lines), 9)
+            self.assertIn("hello", self.multi.lines.mapped('name'))
+        finally:
+            super(One2manyCase, self).tearDown()
```

Comment:
Please don't put the tests in this method. It is not supposed to do that!
Use a simple method that you call in every test method.